src/fft_common.f90

```diff
@@ -186,7 +186,7 @@ subroutine decomp_2d_fft_finalize
 
    if (nx_fft /= nx_global .or. ny_fft /= ny_global .or. nz_fft /= nz_global) then
       call decomp_info_finalize(ph)
-      deallocate (ph)
+      !deallocate (ph)
```

Comment:
Can we use `if (allocated(ph)) deallocate(ph)` instead of commenting the line ?

Reply:
I'll try again but, if i remember well and for reason that are a mystery to me, the `if (allocated(ph))` did not work.

Reply:
if it's a pointer does it need to be `if (associated(ph))`?

Reply:
I need to try but with `associated` the if will be `TRUE` and we'll have again the error on `deallocate`

Reply:
If using `associated` the error persists with the intel compiler, please add a commented line before the commented statement to provide the exact version of the new intel compiler causing the issue.

Reply:
After investigation this isn't necessary - `decomp_info_finalize(ph)` frees the contents of the object, the later call to `nullify` ensures it is no longer pointed to.

Reply:
Please clarify if this is connected with the new or old intel compiler ? Please also specify the exact version of the intel compiler connected with the modification : I think it is a bug and it will probably be fixed later. A similar modification is probably needed in `fftw_f03.f90`.

Reply:
This has been resolved and the `deallocate(ph)` statement is no more necessary